### PR TITLE
Let the forwarders read from the passwords databag

### PR DESCRIPTION
**Pull request by nmattam**
Vagrant provisioning fails on the forwader boxes with the following error.

```
==> f_debian: * template[/opt/splunkforwarder/etc/apps/awesome_splunk_app/default/app.conf] action create
==> f_debian:       
==> f_debian: 
==> f_debian:       
==> f_debian: ================================================================================
==> f_debian:       
==> f_debian: Error executing action `create` on resource 'template[/opt/splunkforwarder/etc/apps/awesome_splunk_app/default/app.conf]'
==> f_debian:       
==> f_debian: ================================================================================
==> f_debian:       
==> f_debian: 
==> f_debian: 
==> f_debian:       
==> f_debian: Chef::Mixin::Template::TemplateError
==> f_debian:       
==> f_debian: ------------------------------------
==> f_debian:       
==> f_debian: cerner_splunk/passwords is not encrypted with your public key.  Contact an administrator of the vault item to encrypt for you!
==> f_debian:       
==> f_debian: 
==> f_debian: 
==> f_debian:       
==> f_debian: Cookbook Trace:
==> f_debian:
```   

This is because _configure_app recipe tries to install the awesome_splunk_app which has the description and label pointing to a vault databag. Forwarder cannot read this vault databag and hence fails.

### CODE REVIEW

File: vagrant_repo/data_bags/cerner_splunk/passwords_keys.json [hunk 1]
```diff
@@ -15,20 +15,28 @@
...
     "c2_deployer",
-    "c2_newnode"
+    "c2_newnode",
+    "f_debian",
+    "f_default",
+    "f_heavy",
+    "f_win2012r2"
   ],
-  "search_query": "name:s_*",
+  "search_query": "name:f_*",
```
Comment (alecxvs) [inline vagrant_repo/data_bags/cerner_splunk/passwords_keys.json]: What does this change do?

Comment (nmattam) [inline vagrant_repo/data_bags/cerner_splunk/passwords_keys.json]: That was the search I used in my knife vault command to add the forwarders.

File: vagrant_repo/data_bags/cerner_splunk/passwords_keys.json [hunk 2]
```diff
@@ -15,20 +15,28 @@
...
+  "knife_workstation": "CVfLEyiNMccDYrReoT7GRDhEkps+eXRTquUFB87hS/FKu7RRVm7yfN5dp61A\n4Ap9r4ykhqapByT02iAZnbB5XqmtZruIrbkZkKrr+iCPrKMzMMs6PtQBGB+g\nMKkWcdJXwAc01R2s2V8BL1YPF/g3kH7izhGprzzjxBwiIVgBzQIMyYGGvVYh\njRQLv/9P1WbEAQNt8tjZhyxTMv5TL83VAJ7+kxAc/G7+i+5kWz2hMNX9RWve\n3ime+7isbTH1Z1jo9KGdUQ+kE0MoV2yaR5yPxUHtku181ww2vLIMd3SPjFdv\nTloeN57wEZ0Rrx4wDQuxlUH7jlBZFWD5MR1NtaBePQ==\n",
   "s_standalone": "zwiC85yUmPJ0q+JaQzrQ6Ug2UsW7F/fBbLiwC26g58i/zbNlS+rvOggsivDV\n0YUs1UZ8rzv/fvM5vfzr3gPLuUYBt2/t9+88GJgcR3zR09E0K77rR0WWsSuo\nT4XAU4pWJM6TsANjnlUCkmgQabHwDhmS3cFxCk3itKiwft0vHWKowj6ObkA4\nGO2udCCX7LrRe8v8o4dLl20Rom3VFNYPD5YlvnMsnrSHoRAsw9EjucLienQ3\nKb4Nvk+DRs2gX6k908G7Zp3tSxx/zHvwmAZrImweNIn5c5F9GzSlVY0rvHo1\n5MZ1Y8IfvA0LqEoMVTLn8ZMY7jivA5ysFVef4Ix3KQ==\n",
   "s_license": "iTEgtfQFH0AuFpdYyoptYSNUkcILZEHS0l8pdGtb+ImaHxuA+rhL1REX+IJK\nA1SxTlTaLgnMC6lre6a2wykWl9H9sXhUUMZGgAwHfuC9vzfWA8s+tFKW58bZ\ncM/2wsqX8e3TNO/lrf0awmvmBKnI1ZErjBzlWcwQFTEl2GycPK0gVnHFz1LZ\nut93D7dT+tJIRR/bpJ5Sx/XtbcWXPcVVlTR02MoHeFA3CR/xnvtMV2qIKymo\noBZo9Emzd3soAwzgoHhWZKTIYrRvtBuRB6IOU1RYL+Jyus08zfqgzSaZzJsg\ngQkPm5s/wuxdPUqwMFLfmAX9++bpaFQ7YdIkNnfnww==\n",
```
Comment (acharlieh) [inline vagrant_repo/data_bags/cerner_splunk/passwords_keys.json]: Given that the value to knife_workstation changed, that means the symmetric key changed, which means that we're broken for all of the other nodes here.

Also given that the symmetric key changed, that means the password data bag item must have changed, and is therefore missing.

Comment (gravesb) [inline vagrant_repo/data_bags/cerner_splunk/passwords_keys.json]: I was able to provision s_standalone and f_default with no issues using this branch of code.

Comment (acharlieh) [inline vagrant_repo/data_bags/cerner_splunk/passwords_keys.json]: Sure enough... using code similar to https://gist.github.com/acharlieh/88e752e9e6b71980778530debe993715 shows that they all decrypt to the same symmetric key value. Looking further into it, the answer is the padding function, where random numbers are included to make the plaintext longer: https://tools.ietf.org/html/rfc2437#section-7.2.1

**Comment by acharlieh** [comment]
+1

**Comment by gravesb** [comment]
+1

**Comment by jdmurphy** [comment]
+1

**Comment by acharlieh** [comment]
Merged without version bump as it is only test code.

**Pull request merged**